File: barbeariainclusao/barbeariainclusao/barbeariainclusao/barbeariainclusao/barbeariainclusao/src/main/java/br/cefetmg/barbeariainclusao/model/Barbearia.java
package br.cefetmg.barbeariainclusao.model;

import lombok.*;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Barbearia {

    private Long idBarbearia;
    private String nomeBarbearia;
    private String enderecoBarbearia;
    private String descricaoBarbearia;
    private String senhaBarbearia;
    private String emailBarbearia;
    private Double latitude;
    private Double longitude;

    // Alterado de String para byte[] para suportar upload de foto
    private byte[] fotoBarbearia;
}

File: BarbeariaEInclusao/src/app/model/barbearia.ts
// Barbearia.ts (Angular/Ionic Model)

export class Barbearia {
    idBarbearia: number;
    nomeBarbearia: string;
    enderecoBarbearia: string;
    descricaoBarbearia: string;
    senhaBarbearia: string;
    emailBarbearia: string;
    fotoBarbearia?: string; // opcional, será Base64

    // NOVOS CAMPOS ADICIONADOS
    latitude?: number;    // Opcional, para armazenar a coordenada
    longitude?: number;   // Opcional, para armazenar a coordenada

    constructor() {
        this.idBarbearia = 0;
        this.nomeBarbearia = '';
        this.enderecoBarbearia = '';
        this.descricaoBarbearia = '';
        this.senhaBarbearia = '';
        this.emailBarbearia = '';
        this.fotoBarbearia = ''; 
        
        // Inicialização opcional dos novos campos
        this.latitude = undefined;
        this.longitude = undefined;
    }
}